src/Model/Model.ts

```diff
@@ -489,16 +489,33 @@ export class Model<T extends IModelData> implements IModel<T> {
     const service = getDataService(this.serviceName);
     observable
       .takeUntil(this.getWillDestroyObservable$())
-      .subscribe((value) => {
+      .subscribe((updatedValue => {
         if (!this.relatedModels.hasOwnProperty(fieldName)) {
-          this.relatedModels[fieldName] = value;
-        } else if (this.relatedModels[fieldName] !== value) {
-          service
-            .actions
-            .setRelationship({ id: this.id, fieldName, value })
-            .invoke();
+          this.relatedModels[fieldName] = updatedValue;
+        } else if (this.relatedModels[fieldName] !== updatedValue) {
+          const currentValue: Model<any> | Model<any>[] | any = this.relatedModels[fieldName];
+
+          if (currentValue instanceof Array) {
+            if (currentValue.some(related => !related.isDestroying)) {
+              currentValue.forEach(related =>
+                // destroy model, so that the subscription above is canceled
+                related.markForDestruction(),
+              );
+              service
```

Comment:
The `setRelatioship` action should be pulled up one level into the `currentValue instanceof Array` if statement. As it's currently written, the related model never get's updated unless at least one of the items in the array isn't marked as `isDestroying`. If you pull the `setRelationship` action up one level then the model get's updated correctly.

Reply:
isnt that exactly what we want?  We don't update relationships when the current value is being destroyed?